### DISCUSSION TITLE
Permission controls for all player interactions that require op

### DISCUSSION THREAD

**Discussion by Roan-V** [PurpurMC/Purpur#434] [Feature Requests]
In extension of #393 
It would be cool if basically anything that requires op could be turned into a permission, e.g. using structure blocks

I currently have no players listed as op on my server, everything is managed through permissions, from time to time I need to use items like the structure block. These can't be used by non-op players. 
A permission to control this would be nice.

## List of things to permissivy \o/ 
> random scrolling person, feel free to add a comment with stuff I forgot
- Place and edit command blocks
would need to verify what permissions the user has, so they can't execute random commands (probably infeasible?)
- Bypass spawn protection
- Place, interact with and use structure blocks
- Place, interact with and use jigsaw blocks (tbh I have no idea wtf these do, so uh, maybe?)

**Already added by purpur 😎:**
- Ability to use the gamemode switcher
- Abiltity to join a full server

**Comment by BillyGalbreath** [comment]
If you can edit this ticket to include a detailed comprehensive list of things you want permissions for I'm sure we can look into it.

**Comment by Roan-V** [comment]
## List of things to permissivy \o/ 
> random scrolling person, feel free to add a comment with stuff I forgot
- Place and edit command blocks
would need to verify what permissions the user has, so they can't execute random commands (probably infeasible?)
- Bypass spawn protection
- Place, interact with and use structure blocks
- Place, interact with and use jigsaw blocks (tbh I have no idea wtf these do, so uh, maybe?)

**Already added by purpur 😎:**
- Ability to use the gamemode switcher
- Abiltity to join a full server

**Comment by emielderckx** [comment]
don't know for sure but a perm for sendCommandFeedback to allow it per player
